### PR TITLE
Add missing external scorer

### PR DESCRIPTION
**Pull request by Jendker**
Bug fix to missing external scorer handling in evaluate_tflite.py

### CODE REVIEW

File: evaluate_tflite.py [hunk 1]
```diff
@@ -29,9 +29,11 @@
 Then run with a TF Lite model, a scorer and a CSV test file
 '''
 
-def tflite_worker(model, scorer, queue_in, queue_out, gpu_mask):
+def tflite_worker(model, scorer, beam_width, queue_in, queue_out, gpu_mask):
     os.environ['CUDA_VISIBLE_DEVICES'] = str(gpu_mask)
     ds = Model(model)
+    ds.enableExternalScorer(scorer)
+    ds.setBeamWidth(beam_width)
```
Comment (lissyx) [inline evaluate_tflite.py:36]: Beam width is stored in the model file, we want to keep `evaluate_tflite.py` as close as possible to use default values from models.

Comment (Jendker) [reply evaluate_tflite.py]: It makes it even simpler then. BeamWidth removed.

File: evaluate_tflite.py [hunk 2]
```diff
@@ -58,7 +59,7 @@ def main(args, _):
 
     processes = []
     for i in range(args.proc):
-        worker_process = Process(target=tflite_worker, args=(args.model, args.scorer, work_todo, work_done, i), daemon=True, name='tflite_process_{}'.format(i))
+        worker_process = Process(target=tflite_worker, args=(args.model, args.scorer, args.beam_width, work_todo, work_done, i), daemon=True, name='tflite_process_{}'.format(i))
```
Comment (lissyx) [inline evaluate_tflite.py:62]: nit: you forgot to remove it

**Comment by lissyx** [comment]
Thanks @Jendker. Can you ensure it works as expected?

**Comment by lissyx** [comment]
Also, squash into one commit :)

**Comment by Jendker** [comment]
I squashed the commits and pushed force. Is it the correct workflow?

**Comment by lissyx** [comment]
> I squashed the commits and pushed force. Is it the correct workflow?

Looks good :)

**Comment by Jendker** [comment]
I'll verify if it works as expected soon and report if it doesn't.

**Comment by lissyx** [comment]
@Jendker I'd like to wait for your go one you have been able to confirm it works :)

**Comment by Jendker** [comment]
Right, I'll come back in a couple of hours/tomorrow.

**Comment by lissyx** [comment]
> Right, I'll come back in a couple of hours/tomorrow.

So I'm sure it fixed the lack of scorer, but @Jendker is it working correctly ? Is it enough for fixing your WER issues ?

**Comment by Jendker** [comment]
Sorry for the slow answer.

WER is very similar right now:
evaluate.py - 0.192821
evaluate_tflite.py - 0.191987

so I think the problem is solved :)
The only issue which I found is that if I specify the scorer to non-existing path I get very cryptic error:
```
python evaluate_tflite.py --model "/data/DeepSpeech/deepspeech-cleaner/languages/de/training/youtube_eml_till_2020_Feb_transfer_learning_fat_data_no_LM_for_training_cont_lr_0.00001/model_export/output_graph.pb" --scorer /data/DeepSpeech/DS_worker_path/model/kenlm_ds.scorer --csv /home/ben/test.csv --proc 1
['tflite_process_0']
Totally 99 wav entries found in csv

TensorFlow: v1.15.0-24-gceb46aa
DeepSpeech: v0.7.0-alpha.3-0-g58bc2f2
Warning: reading entire model file into memory. Transform model file into an mmapped graph to reduce heap usage.
2020-04-21 11:55:52.642798: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
Fatal Python error: Segmentation fault

Current thread 0x00007fa1a1af7740 (most recent call first):
  File "/home/ben/DeepSpeech/venv/lib/python3.6/site-packages/deepspeech/impl.py", line 158 in SpeechToText
  File "/home/ben/DeepSpeech/venv/lib/python3.6/site-packages/deepspeech/__init__.py", line 122 in stt
  File "evaluate_tflite.py", line 46 in tflite_worker
  File "/usr/lib/python3.6/multiprocessing/process.py", line 93 in run
  File "/usr/lib/python3.6/multiprocessing/process.py", line 258 in _bootstrap
  File "/usr/lib/python3.6/multiprocessing/popen_fork.py", line 73 in _launch
  File "/usr/lib/python3.6/multiprocessing/popen_fork.py", line 19 in __init__
  File "/usr/lib/python3.6/multiprocessing/context.py", line 277 in _Popen
  File "/usr/lib/python3.6/multiprocessing/context.py", line 223 in _Popen
  File "/usr/lib/python3.6/multiprocessing/process.py", line 105 in start
  File "evaluate_tflite.py", line 63 in main
  File "/home/ben/DeepSpeech/venv/lib/python3.6/site-packages/absl/app.py", line 250 in _run_main
  File "/home/ben/DeepSpeech/venv/lib/python3.6/site-packages/absl/app.py", line 299 in run
  File "evaluate_tflite.py", line 126 in <module>
^CTraceback (most recent call last):
  File "evaluate_tflite.py", line 126, in <module>
    absl.app.run(partial(main, parse_args()))
  File "/home/ben/DeepSpeech/venv/lib/python3.6/site-packages/absl/app.py", line 299, in run
    _run_main(main, args)
  File "/home/ben/DeepSpeech/venv/lib/python3.6/site-packages/absl/app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "evaluate_tflite.py", line 86, in main
    work_todo.join()
  File "/usr/lib/python3.6/multiprocessing/queues.py", line 305, in join
    self._cond.wait()
  File "/usr/lib/python3.6/multiprocessing/synchronize.py", line 261, in wait
    return self._wait_semaphore.acquire(True, timeout)
KeyboardInterrupt
```
It would be good to add some logic to check the scorer path. Is there any function which would check the scorer itself which I could add?

**Comment by lissyx** [comment]
> so I think the problem is solved :)

Thanks!



> It would be good to add some logic to check the scorer path. Is there any function which would check the scorer itself which I could use? Or should I just check if file exists?

I think in other places we just try and load the scorer and let it fail this way. Fixing it properly is likely to be done in the `native_client/ctcdecoder` part, if you want to have a look.

**Pull request merged**